pkg/admission/interfaces.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+)
+
+// Attributes is an interface used by AdmissionController to get information about a request
+// that is used to make an admission decision.
+type Attributes interface {
+	GetClient() client.Interface
+	GetNamespace() string
+	GetKind() string
+	GetOperation() string
+	GetObject() runtime.Object
+}
+
+// Interface is an abstract, pluggable interface for Admission Control decisions.
+type Interface interface {
+	// Admit makes an admission decision based on the request attributes
+	Admit(a Attributes) (err error)
+}
+
```

Comment:
The major difference between the interfaces is that one is expected to have a reference to a suitable client object to pass in as ```Attributes``` to the implementation of admission.Interface.  Its possible that an ```AdmissionControl``` object has some intelligence of giving the right client object for a particular request kind.  I was mostly thinking that I may need different clients when working in an OpenShift context than core Kube depending on the resource, and this was my hook to avoid passing down multiple clients through the resthandler.
Major difference between the implementations is that one holds onto a client object and therefore knows how to build the ```Attributes``` object.  Its possible this could be improved because I see your point.  The